Mathlib/NumberTheory/NumberField/CanonicalEmbedding/PolarCoord.lean

```diff
@@ -0,0 +1,334 @@
+    (FDerivPolarCoordRealSymm K x).det = ∏ w : {w // IsComplex w}, (x.2 w).1 := by
+  have : (FDerivPolarCoordRealSymm K x).toLinearMap =
+      LinearMap.prodMap (LinearMap.id) (fderivPiPolarCoordSymm x.2).toLinearMap := rfl
+  rw [ContinuousLinearMap.det, this, LinearMap.det_prodMap, LinearMap.det_id, one_mul,
+    ← ContinuousLinearMap.det, det_fderivPiPolarCoordSymm]
+
+open scoped Classical in
+theorem polarCoordReal_symm_target_ae_eq_univ :
+    (polarCoordReal K).symm '' (polarCoordReal K).target =ᵐ[volume] Set.univ := by
+  rw [← Set.univ_prod_univ, volume_eq_prod, (polarCoordReal K).symm_image_target_eq_source,
+    polarCoordReal_source, ← polarCoord.symm_image_target_eq_source, ← Set.piMap_image_univ_pi]
+  exact set_prod_ae_eq (by rfl) pi_polarCoord_symm_target_ae_eq_univ
+
+open scoped Classical in
+theorem integral_comp_polarCoordReal_symm  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+    (f : realMixedSpace K → E) :
+    ∫ x in (polarCoordReal K).target, (∏ w : {w // IsComplex w}, (x.2 w).1) •
+      f ((polarCoordReal K).symm x) = ∫ x, f x := by
+  rw [← setIntegral_univ (f := f), ← setIntegral_congr_set
+    (polarCoordReal_symm_target_ae_eq_univ K), integral_image_eq_integral_abs_det_fderiv_smul
+    volume (polarCoordReal K).open_target.measurableSet
+    (fun x _ ↦ (hasFDerivAt_polarCoordReal_symm K x).hasFDerivWithinAt)
+    (polarCoordReal K).symm.injOn f]
+  refine setIntegral_congr_fun (polarCoordReal K).open_target.measurableSet fun x hx ↦ ?_
+  simp_rw [det_fderivPolarCoordRealSymm, Finset.abs_prod, abs_of_mem_polarCoordReal_target K hx]
+
+open scoped Classical in
+theorem lintegral_comp_polarCoordReal_symm (f : realMixedSpace K → ℝ≥0∞) :
+    ∫⁻ x in (polarCoordReal K).target, (∏ w : {w // IsComplex w}, .ofReal (x.2 w).1) *
+      f ((polarCoordReal K).symm x) = ∫⁻ x, f x := by
+  rw [← setLIntegral_univ f, ← setLIntegral_congr (polarCoordReal_symm_target_ae_eq_univ K),
+    lintegral_image_eq_lintegral_abs_det_fderiv_mul volume
+    (polarCoordReal K).open_target.measurableSet
+    (fun x _ ↦ (hasFDerivAt_polarCoordReal_symm K x).hasFDerivWithinAt)
+    (polarCoordReal K).symm.injOn f]
+  refine setLIntegral_congr_fun (polarCoordReal K).open_target.measurableSet ?_
+  filter_upwards with x hx
+  simp_rw [det_fderivPolarCoordRealSymm, Finset.abs_prod, ENNReal.ofReal_prod_of_nonneg (fun _ _ ↦
+    abs_nonneg _), abs_of_mem_polarCoordReal_target K hx]
+
+end realMixedSpace
+
+section mixedSpace
+
+variable [NumberField K]
+
+/--
+The polar coordinate partial homeomorphism between the mixed space `ℝ^r₁ × ℂ^r₂` and
+`ℝ^r₁ × (ℝ × ℝ)^r₂` defined as the identity on the first component and mapping `(zᵢ)ᵢ` to
+`(‖zᵢ‖, Arg zᵢ)ᵢ` on the second component.
+-/
+@[simps!]
+protected noncomputable def polarCoord : PartialHomeomorph (mixedSpace K) (realMixedSpace K) :=
+  (PartialHomeomorph.refl _).prod (PartialHomeomorph.pi fun _ ↦ Complex.polarCoord)
+
+theorem polarCoord_target_eq_polarCoordReal_target :
```

Comment:
Maybe this and the next one could be `simp`.

Reply:
In fact, it's better if they are not: lint error for one and breaking some proof for the other one.

Thanks for the review!